contracts/helpers/order-helper/lib/OrderStructureLib.sol

```diff
@@ -0,0 +1,516 @@
+
+        // If any non-contract item has criteria, it's advanced,
+        if (hasCriteria) {
+            // Unless it's a contract order
+            if (isContractOrder) {
+                // And the contract order's critera are all zero
+                if (hasNonzeroCriteria) {
+                    return Structure.ADVANCED;
+                }
+            } else {
+                return Structure.ADVANCED;
+            }
+        }
+
+        if (getBasicOrderTypeEligibility(order, seaport)) {
+            return Structure.BASIC;
+        }
+
+        return Structure.STANDARD;
+    }
+
+    function getStructure(
+        AdvancedOrder[] memory orders,
+        address seaport
+    ) internal view returns (Structure) {
+        if (orders.length == 1) {
+            return getStructure(orders[0], seaport);
+        }
+
+        for (uint256 i; i < orders.length; i++) {
+            Structure structure = getStructure(orders[i], seaport);
+            if (structure == Structure.ADVANCED) {
+                return Structure.ADVANCED;
+            }
+        }
+
+        return Structure.STANDARD;
+    }
+
+    function getBasicOrderTypeEligibility(
+        AdvancedOrder memory order,
+        address seaport
+    ) internal view returns (bool) {
+        uint256 i;
+        ConsiderationItem[] memory consideration = order
+            .parameters
+            .consideration;
+        OfferItem[] memory offer = order.parameters.offer;
+
+        // Order must contain exactly one offer item and one or more
+        // consideration items.
+        if (offer.length != 1) {
+            return false;
+        }
+        if (
+            consideration.length == 0 ||
+            order.parameters.totalOriginalConsiderationItems == 0
+        ) {
+            return false;
+        }
+
+        // The order cannot have a contract order type.
+        if (order.parameters.orderType == OrderType.CONTRACT) {
+            return false;
+
+            // Note: the order type is combined with the “route” into a single
+            // BasicOrderType with a value between 0 and 23; there are 4
+            // supported order types (full open, partial open, full restricted,
+            // partial restricted) and 6 routes (ETH ⇒ ERC721, ETH ⇒ ERC1155,
+            // ERC20 ⇒ ERC721, ERC20 ⇒ ERC1155, ERC721 ⇒ ERC20, ERC1155 ⇒ ERC20)
+        }
+
+        // Order cannot specify a partial fraction to fill.
+        if (order.denominator > 1 && (order.numerator < order.denominator)) {
+            return false;
+        }
+
+        // Order cannot be partially filled.
+        ConsiderationInterface seaportInterface = ConsiderationInterface(
+            seaport
+        );
+        uint256 counter = seaportInterface.getCounter(order.parameters.offerer);
+        OrderComponents memory orderComponents = order
+            .parameters
+            .toOrderComponents(counter);
+        bytes32 orderHash = seaportInterface.getOrderHash(orderComponents);
```

Comment:
shouldn't this be operating on the "tip-neutralized" order hash?

Reply:
Interesting. Should this behave differently than the [fuzz engine](https://github.com/ProjectOpenSea/seaport/blob/cbcd738ee7be5c4178d550b843b71c0bb18d2d04/test/foundry/new/helpers/FuzzHelpers.sol#L510)? It's also using plain order hash rather than the tip neutralized version.

Reply:
might honestly be a "bug" in the fuzz engine (causing us to not pick basic orders as often as we could)?